mindsdb/integrations/handlers/replicate_handler/replicate_handler.py

```diff
@@ -0,0 +1,145 @@
+            if e.args[0].startswith('Not found'):
+                raise Exception(f"Could not retrieve version {args['version']} of model {args['model_name']}. Verify values are correct and try again.", e)
+
+            elif e.args[0].startswith('Incorrect authentication token'):
+                raise Exception("Provided api_key is Incorrect. Get your api_key here: https://replicate.com/account/api-tokens")
+
+            else:
+                raise Exception(f"Error occured.", e)
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """Saves model details in storage to access it later
+        """
+        args = args['using']
+        args['target'] = target
+        self.model_storage.json_set('args', args)
+
+    def predict(self, df: pd.DataFrame, args: Optional[Dict] = None) -> pd.DataFrame:
+        """Using replicate makes the prediction according to your parameters
+        """
+        def get_data(conditions):
+            output = replicate.run(
+                f"{args['model_name']}:{args['version']}",
+                input={**conditions[1].to_dict(), **pred_args})  # unpacking parameters inputted
+
+            if isinstance(output, types.GeneratorType) and args.get('model_type') == 'LLM':
+                output = [''.join(list(output))]      # If model_type is LLM make stream as string
+            elif isinstance(output, types.GeneratorType):
+                output = [list(output)[-1]]         # Getting final url if output is generator of frames url
+            elif isinstance(output, list) and len(output) > 1:
+                output = [output[-1]]             # returns generated image for controlNet models as it outputs filter and generated image
+            return output
+
+        pred_args = args['predict_params'] if args else {}
+        args = self.model_storage.json_get('args')
+        model_name, version, target_col = args['model_name'], args['version'], args['target']
+        # raising Exception if wrong parameters is given
+        params_names = set(df.columns) | set(pred_args)
+        available_params = self._get_schema(only_keys=True)
+        wrong_params = []
+        for i in params_names:
+            if i not in available_params:
+                wrong_params.append(i)
+
+        if wrong_params:
+            raise Exception(f"""'{wrong_params}' is/are not supported parameter for this model.
+Use DESCRIBE PREDICTOR mindsdb.<model_name>.features; to know about available parameters. OR 
+Visit https://replicate.com/{model_name}/versions/{version} to check parameters.""")
+
+        replicate.default_client.api_token = self._get_replicate_api_key(args)
+
+        rows = df.iterrows()      # Generator is returned
+        rows_arr = list(rows)      # Generator converted to list have tuple consisting (index, row(pandas.Series))
+        data = map(get_data, rows_arr)  # using get_data function to get data according to inputted parameters
```

Comment:
I think (may be wrong though) that doing `df.apply(get_data)` should be faster than this `list(iterrows())` approach, seems worth trying out because `iterrows()` is really slow. It would require a slight change in the way you're indexing data coming in at `get_data`, but nothing major.

Reply:
done